src/utils/fragment-tools.ts

```diff
@@ -84,6 +84,74 @@ export const hasFragmentStyleClasses = (fragment: any) => {
 	return hasComponentStyleClasses(fragment.data);
 };
 
+export const getUniqueFragmentName = (fragments: Array<any>, baseName: string) => {
+	const nameRegEx = new RegExp(String.raw`(.*)\s+(copy)*(\s+(\d+))?$`);
+	const nameMatch = baseName.match(nameRegEx);
+	let count = 0;
+
+	let nameBase = baseName;
+	// If match, increment the count and update name base and new name
+	if (nameMatch) {
+		nameBase = baseName.replace(nameRegEx, '$1');
+		count = Number.parseInt(baseName.replace(nameRegEx, '$4'), 10);
+		if (!count) {
+			count = 0;
+		}
+	}
+
+	// Get a list containing names of all duplicates of original fragment
+	// e.g. [ "Fragment copy", "Fragment copy 1", "Fragment copy 7", ...]
+	const names: string[] = [];
+	fragments.forEach((fragment) => {
+		if (fragment.title.includes(nameBase)) {
+			names.push(fragment.title);
+		}
+	});
+
+	if (names.length <= 1) {
+		// because the fragment we're copying is already in there
+		return `${nameBase} copy`;
+	}
+
+	const highestNumber = names
+		.map((n) => Number.parseInt(n.replace(nameRegEx, '$4'), 10))
+		.filter((n) => !isNaN(n)).sort((a, b) => b - a)
+		.shift();
+
+	return `${nameBase} copy ${highestNumber && count < highestNumber ? highestNumber + 1 : count + 1}`;
+};
+
+export const duplicateFragment = (fragments: any, fragment: any, overrides = {}) => {
```

Comment:
`getFragmentDuplicate` or `newFromFragment` or something similar?

I kinda expected it to take in setFragments and add it to the list, based on the name.